generators/client/templates/angular/src/main/webapp/app/blocks/interceptor/_auth.interceptor.ts

```diff
@@ -17,34 +17,32 @@
  limitations under the License.
 -%>
 import { Observable } from 'rxjs/Observable';
-import { RequestOptionsArgs, Response } from '@angular/http';
 import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
-import { JhiHttpInterceptor } from 'ng-jhipster';
+import { HttpInterceptor, HttpRequest, HttpHandler, HttpEvent } from '@angular/common/http';
 import { SERVER_API_URL } from '../../app.constants';
 
-export class AuthInterceptor extends JhiHttpInterceptor {
+export class AuthInterceptor implements HttpInterceptor {
 
     constructor(
         private localStorage: LocalStorageService,
         private sessionStorage: SessionStorageService
     ) {
-        super();
     }
 
-    requestIntercept(options?: RequestOptionsArgs): RequestOptionsArgs {
-        if (!options || !options.url || (/^http/.test(options.url) && !(SERVER_API_URL && options.url.startsWith(SERVER_API_URL)))) {
-            return options;
+    intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        if (!request || !request.url || (/^http/.test(request.url) && !(SERVER_API_URL && request.url.startsWith(SERVER_API_URL)))) {
+            return next.handle(request);
         }
 
         const token = this.localStorage.retrieve('authenticationToken') || this.sessionStorage.retrieve('authenticationToken');
         if (!!token) {
-            options.headers.append('Authorization', 'Bearer ' + token);
+            request = request.clone({
```

Comment:
Doesn't it mutate the entire headers? If there are any headers set?

Reply:
no according to this implementation https://github.com/angular/angular/blob/5.1.3/packages/common/http/src/request.ts#L67-L385 

setHeaders adds the header. headers would mutate all headers. 

badnaming on angular site though